be/src/storage/object_metastore.cpp

```diff
@@ -0,0 +1,198 @@
+    ASSIGN_OR_RETURN(auto objects, list_objects_by_prefix(object_prefix));
+    std::vector<RowsetMetaSharedPtr> rowset_metas;
+    rowset_metas.reserve(objects.size());
+    for (const auto& object : objects) {
+        auto path = fmt::format("{}/{}", _meta_dir, object);
+        auto rowset_meta_st = get_rowset_meta(path);
+        if (!rowset_meta_st.ok()) {
+            return rowset_meta_st.status();
+        }
+        rowset_metas.emplace_back(rowset_meta_st.value());
+    }
+    return std::move(rowset_metas);
+}
+
+Status ObjectMetastore::remove_rowset_meta(const TabletUid& tablet_uid, const RowsetId& rowset_id) {
+    auto path = fmt::format("{}/{}{}_{}", _meta_dir, kRowsetMetaPrefix, tablet_uid.to_string(), rowset_id.to_string());
+    return delete_object(path);
+}
+
+Status ObjectMetastore::put_object(const std::string& path, const std::string& object) {
+    ASSIGN_OR_RETURN(auto env, Env::CreateUniqueFromString(path));
+    ASSIGN_OR_RETURN(auto file, env->new_writable_file(path));
+    RETURN_IF_ERROR(file->append(Slice(object)));
+    RETURN_IF_ERROR(file->sync());
+    return file->close();
+}
+
+StatusOr<std::string> ObjectMetastore::get_object(const std::string& path) {
+    ASSIGN_OR_RETURN(auto env, Env::CreateUniqueFromString(path));
+    ASSIGN_OR_RETURN(auto file, env->new_random_access_file(path));
+    ASSIGN_OR_RETURN(auto size, file->get_size());
+    std::vector<char> content;
+    raw::stl_vector_resize_uninitialized(&content, size);
+    RETURN_IF_ERROR(file->read_at_fully(0, content.data(), size));
+    return std::string(content.data(), content.size());
+}
+
+StatusOr<std::vector<std::string>> ObjectMetastore::list_objects_by_prefix(const std::string& prefix_path) {
+    S3URI uri;
+    if (!uri.parse(prefix_path)) {
+        return Status::InvalidArgument(fmt::format("Invalid S3 URI {}", prefix_path));
+    }
+
+    std::vector<std::string> objects;
+    auto client = new_s3client(uri);
+    Aws::S3::Model::ListObjectsV2Request request;
+    Aws::S3::Model::ListObjectsV2Result result;
+    request.WithBucket(uri.bucket()).WithPrefix(uri.key()).WithDelimiter("/");
+    do {
+        auto outcome = client->ListObjectsV2(request);
+        if (!outcome.IsSuccess()) {
+            return Status::IOError(
+                    fmt::format("S3: fail to list {}: {}", prefix_path, outcome.GetError().GetMessage()));
+        }
+        result = outcome.GetResultWithOwnership();
+        request.SetContinuationToken(result.GetNextContinuationToken());
+        // Get file objects except directory.
+        for (auto&& obj : result.GetContents()) {
+            DCHECK(HasPrefixString(obj.GetKey(), uri.key()));
+            std::string_view key = obj.GetKey();
+            size_t pos = key.find_last_of("/");
+            if (pos == std::string::npos) {
+                objects.emplace_back(key);
+            } else {
+                objects.emplace_back(key.substr(pos + 1));
```

Comment:
Since only `get_rowset_metas()` will call this method, instead of saving the object URI into the vector, maybe we can call `get_rowset_meta(obj.GetKey())` directly here to fetch the rowset meta.

Reply:
Maybe also used to list tablet metas later.